### ISSUE TITLE
Сhange the size of the object along "Y" for visualizing the liquid in the tank

### ISSUE THREAD

**Issue by Sledok** [alex-controlx/red-dashboard-svg-control#30]
Hi Alex! can you help me I want to make a mini SCADA in the NODE RED environment. i liked your code. it almost works for me. but there was a problem in solving my tasks. I need to change the size of the object along "Y" for visualizations of the liquid in the tank. Thank you in advance for your help and sorry for the language, I use a translator. Because I'm from Ukraine
[flows .txt](https://github.com/alex-controlx/red-dashboard-svg-control/files/12789625/flows.txt)
![Снимок экрана (17)](https://github.com/alex-controlx/red-dashboard-svg-control/assets/66492522/4d87577b-1103-4459-926c-8e3016b349b6)

**Comment by alex-controlx** [comment]
This sounds like a feature request. I'll keep this open and get to this once I can clearly understand the way to implement it.

In the meantime you can fix it with a simple hack. Under the tank create a square shape with the same colour as the background. The "liquid" in the tank need to be at the lower layer than the square. So if you move the "liquid" down it will be hidden under the square. Now you can use the `move` class to move the "liquid" along Y and it will appear as the tank gets full.

**Comment by Sledok** [comment]
As I understood, the solution to this quest will not be the same for all objects. Because there are, as I noticed, two formats for recording points vertically and horizontally. 1. width="220" height="300" Or 2.d="M -1158.042 -715.812 L -1149.042 -687.812 L -1167.042 -687.812 L -1158.042 -715.812 Z"

**Comment by alex-controlx** [comment]
Yes, to add this feature it will be needed to consider all ways an object can be transformed. I'll read a bit on this topic.

**Comment by Sledok** [comment]
Hello Alex. I did as you told me, insert a picture so that my object hides behind the picture and there was an effect of changing the level in the tanker. Looks pretty good.

**Comment by webbrain-one** [comment]
Hi! 👋

I've opened [PR #33](https://github.com/alex-controlx/red-dashboard-svg-control/pull/33) to address this. It adds the ability to change the size of the object along the Y-axis, which should allow you to resize your tank for liquid visualization in your Node-RED SCADA project.

Let me know if you have any questions or need anything else. Best of luck with your project! 🚀

**Comment by Sledok** [comment]
Hi Alex! I’ve been a bit busy with other work lately, but I still find this
project really interesting. I’ll definitely give it a try when I have some
free time. Thanks for sharing it with me! How have you been? How are things
going on your side?

ср, 5 серп. 2026 р. о 08:30 WebBrain ***@***.***> пише:
